tests/io/test_file_read.py

```diff
@@ -0,0 +1,149 @@
+import time
+
+import h5py as h5
+import numpy as np
+
+from miv.io import file as miv_file
+
+
+def isEmpty(dictionary):
+    test = True
+    print(dictionary.keys())
+    for key in dictionary.keys():
+        print(key)
+        print(dictionary[key])
+        print(type(dictionary[key]))
+        if dictionary[key] is None:
+            test = True
+        elif type(dictionary[key]) == list or type(dictionary[key]) == np.ndarray:
+            if len(dictionary[key]) > 0:
+                test = False
+
+    return test
+
+
+def test_write_file():
+
+    data = miv_file.initialize()
+
+    miv_file.create_group(data, "coordinates", counter="ncoords")
+    miv_file.create_dataset(data, ["px", "py", "pz"], group="coordinates", dtype=float)
+
+    miv_file.create_dataset(data, ["u", "v"], group="electrodes", dtype=float)
+
+    event = miv_file.create_container(data)
+
+    for i in range(0, 10):
+
+        ncoords = 5
+        event["coordinates/ncoords"] = ncoords
+
+        for n in range(ncoords):
+            event["coordinates/px"].append(np.random.random())
+            event["coordinates/py"].append(np.random.random())
+            event["coordinates/pz"].append(np.random.random())
+
+        event["electrodes/u"].append(np.random.random())
+        event["electrodes/v"].append(np.random.random())
+
+        miv_file.pack(data, event)
+
+    miv_file.write("MiV_TESTS.h5", data, comp_type="gzip", comp_opts=9)
+
+
+def test_read():
+
+    test_write_file()
+
+    filename = "MiV_TESTS.h5"
+    desired_datasets = ["coordinates", "electrodes"]
+    subset = 5
+
+    test_data, test_container = miv_file.read(filename, desired_datasets, subset)
+
+    assert isinstance(test_data, dict)
+    assert isinstance(test_container, dict)
+
+    assert isEmpty(test_container)
+    assert not isEmpty(test_data)
+
+    # Testing desired_datasets
+    assert "coordinates/px" in test_data.keys()
+
+    # Testing subsets
+    assert len(test_data["coordinates/ncoords"]) == 5
```

Comment:
@iraikov This one test keeps failing.

Could you double check the `subset` implementation?

Reply:
I think 9430043 this should fix the issue.